backend/de.metas.payment.esr/src/main/java/de/metas/payment/esr/dataimporter/impl/camt54/ReferenceStringHelper.java

```diff
@@ -138,11 +170,12 @@ public void extractAndSetEsrReference(
 		final Optional<String> esrReferenceNumberString = txDtls.getRmtInf().getStrd().stream()
 				.map(strd -> strd.getCdtrRefInf())
 
-				// it's stored in the cdtrRefInf records whose cdtrRefInf/tp/cdOrPrtry/prtr equals to ISR_REFERENCE
-				.filter(cdtrRefInf -> cdtrRefInf != null
-						&& cdtrRefInf.getTp() != null
+				// it's stored in the cdtrRefInf records whose cdtrRefInf/tp/cdOrPrtry/prtr equals to ISR_REFERENCE or QRR_REFERENCE
+				.filter(cdtrRefInf -> cdtrRefInf != null 
```

Comment:
i guess u can extract in a method the whole expression from filter(...).
Saw it that u are using it in couple of places.